core/src/main/java/bisq/core/dao/monitoring/ProposalStateMonitoringService.java

```diff
@@ -134,15 +134,18 @@ public void onDaoStateChanged(Block block) {
         int blockHeight = block.getHeight();
         int genesisBlockHeight = genesisTxInfo.getGenesisBlockHeight();
 
+        boolean hashChainUpdated = false;
         if (proposalStateBlockChain.isEmpty() && blockHeight > genesisBlockHeight) {
             // Takes about 150 ms for dao testnet data
             long ts = System.currentTimeMillis();
             for (int i = genesisBlockHeight; i < blockHeight; i++) {
-                maybeUpdateHashChain(i);
+                hashChainUpdated = hashChainUpdated || maybeUpdateHashChain(i);
```

Comment:
Does this not have the unintended consequence of only updating the hashchain for the first cycle? Since once maybeUpdateHashChain returns true (first block of blind vote phase), this statement will short-circuit and not call maybeUpdateHashChain for any other blocks?

Reply:
Never mind. Just saw your subsequent commit :)